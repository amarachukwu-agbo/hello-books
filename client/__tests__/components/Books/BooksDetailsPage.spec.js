import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedBookDetailsPage, { BookDetailsPage } from
  '../../../components/Books/BookDetailsPage';

const props = {
  getBook: sinon.spy(() => Promise.resolve()),
  favoriteBook: sinon.spy(() => Promise.resolve()),
  upvoteBook: sinon.spy(() => Promise.resolve()),
  borrowBook: sinon.spy(() => Promise.resolve()),
  reviewBook: sinon.spy(() => Promise.resolve()),
  isFetching: false,
  getBookError: null,
  match: {
    params: {
      bookId: 97,
    },
  },
  book: {
    id: 98,
    title: 'I"ll Be Gone in the Dark',
    author: 'Michelle McNamara',
    description: 'A masterful true crime account of the Golden State Killer.',
    subject: 'True Crime',
    imageURL: 'https://res/adolfo-felix-586654-unsplash_ukhitd.jpg',
    quantity: 3,
    borrowCount: 6,
    favCount: 3,
    upvotes: 1,
    downvotes: 4,
    createdAt: '2018-05-10T03:11:52.181Z',
    updatedAt: '2018-07-10T15:46:02.771Z',
    bookReviews: [
      {
        id: 20,
        review: 'Review',
        createdAt: '2018-07-10T08:51:25.310Z',
        updatedAt: '2018-07-10T08:51:25.310Z',
        userId: 16,
        bookId: 98,
        userReviews: {
          id: 16,
          firstName: 'Agbo',
          lastName: 'Tochi',
          email: 'rtochi57@gmail.com',
          role: 'User',
          imageURL: null,
          createdAt: '2018-07-08T19:29:20.334Z',
          updatedAt: '2018-07-08T19:29:20.334Z',
        },
      },
    ],
  },
};

const mockStore = configureMockStore([thunk]);
const store = mockStore({});

describe('<BooksDetailsPage />', () => {
  sinon.spy(BookDetailsPage.prototype, 'componentDidMount');
  it('renders correctly', () => {
    const wrapper = shallow(<BookDetailsPage {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('calls componentDidMount()', () => {
    shallow(<BookDetailsPage {...props} />);
    expect(BookDetailsPage.prototype.componentDidMount.called)
      .toEqual(true);
    expect(props.getBook.called).toEqual(true);
    expect(props.getBook.calledWith(97)).toEqual(true);
  });
  it('calls render preloader when book is fetching', () => {
    props.isFetching = true;
    const wrapper = shallow(<BookDetailsPage {...props}/>);
    expect(wrapper.find('Preloader')).toHaveLength(1);
  });
  it('renders error when there is an error fetching book', () => {
    props.isFetching = false;
    props.getBookError = 'Book was not found';
    const wrapper = shallow(<BookDetailsPage {...props}/>);
    expect(wrapper.find('h5').text())
      .toEqual('The book you requested could not be retrieved. Book was not found'); /*eslint-disable-line*/
  });
});

describe('<ConnectedBooksDetailsPage/>', () => {
  const connectedWrapper = shallow(<ConnectedBookDetailsPage
    store = {store} {...props}/>);
  expect(connectedWrapper.length).toBe(1);
});
