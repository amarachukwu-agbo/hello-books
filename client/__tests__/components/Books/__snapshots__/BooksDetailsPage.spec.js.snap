// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BooksDetailsPage /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookDetailsPage
    book={
      Object {
        "author": "Michelle McNamara",
        "bookReviews": Array [
          Object {
            "bookId": 98,
            "createdAt": "2018-07-10T08:51:25.310Z",
            "id": 20,
            "review": "Review",
            "updatedAt": "2018-07-10T08:51:25.310Z",
            "userId": 16,
            "userReviews": Object {
              "createdAt": "2018-07-08T19:29:20.334Z",
              "email": "rtochi57@gmail.com",
              "firstName": "Agbo",
              "id": 16,
              "imageURL": null,
              "lastName": "Tochi",
              "role": "User",
              "updatedAt": "2018-07-08T19:29:20.334Z",
            },
          },
        ],
        "borrowCount": 6,
        "createdAt": "2018-05-10T03:11:52.181Z",
        "description": "A masterful true crime account of the Golden State Killer.",
        "downvotes": 4,
        "favCount": 3,
        "id": 98,
        "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
        "quantity": 3,
        "subject": "True Crime",
        "title": "I\\"ll Be Gone in the Dark",
        "updatedAt": "2018-07-10T15:46:02.771Z",
        "upvotes": 1,
      }
    }
    borrowBook={[Function]}
    favoriteBook={[Function]}
    getBook={[Function]}
    getBookError={null}
    isFetching={false}
    match={
      Object {
        "params": Object {
          "bookId": 97,
        },
      }
    }
    reviewBook={[Function]}
    upvoteBook={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
        <div
          id="book-details"
        >
          <BookDetails
            book={
              Object {
                "author": "Michelle McNamara",
                "bookReviews": Array [
                  Object {
                    "bookId": 98,
                    "createdAt": "2018-07-10T08:51:25.310Z",
                    "id": 20,
                    "review": "Review",
                    "updatedAt": "2018-07-10T08:51:25.310Z",
                    "userId": 16,
                    "userReviews": Object {
                      "createdAt": "2018-07-08T19:29:20.334Z",
                      "email": "rtochi57@gmail.com",
                      "firstName": "Agbo",
                      "id": 16,
                      "imageURL": null,
                      "lastName": "Tochi",
                      "role": "User",
                      "updatedAt": "2018-07-08T19:29:20.334Z",
                    },
                  },
                ],
                "borrowCount": 6,
                "createdAt": "2018-05-10T03:11:52.181Z",
                "description": "A masterful true crime account of the Golden State Killer.",
                "downvotes": 4,
                "favCount": 3,
                "id": 98,
                "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                "quantity": 3,
                "subject": "True Crime",
                "title": "I\\"ll Be Gone in the Dark",
                "updatedAt": "2018-07-10T15:46:02.771Z",
                "upvotes": 1,
              }
            }
            borrowBook={[Function]}
            favoriteBook={[Function]}
            getBook={[Function]}
            getBookError={null}
            isFetching={false}
            match={
              Object {
                "params": Object {
                  "bookId": 97,
                },
              }
            }
            reviewBook={[Function]}
            upvoteBook={[Function]}
          />
          <br />
          <Reviews
            book={
              Object {
                "author": "Michelle McNamara",
                "bookReviews": Array [
                  Object {
                    "bookId": 98,
                    "createdAt": "2018-07-10T08:51:25.310Z",
                    "id": 20,
                    "review": "Review",
                    "updatedAt": "2018-07-10T08:51:25.310Z",
                    "userId": 16,
                    "userReviews": Object {
                      "createdAt": "2018-07-08T19:29:20.334Z",
                      "email": "rtochi57@gmail.com",
                      "firstName": "Agbo",
                      "id": 16,
                      "imageURL": null,
                      "lastName": "Tochi",
                      "role": "User",
                      "updatedAt": "2018-07-08T19:29:20.334Z",
                    },
                  },
                ],
                "borrowCount": 6,
                "createdAt": "2018-05-10T03:11:52.181Z",
                "description": "A masterful true crime account of the Golden State Killer.",
                "downvotes": 4,
                "favCount": 3,
                "id": 98,
                "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                "quantity": 3,
                "subject": "True Crime",
                "title": "I\\"ll Be Gone in the Dark",
                "updatedAt": "2018-07-10T15:46:02.771Z",
                "upvotes": 1,
              }
            }
            borrowBook={[Function]}
            favoriteBook={[Function]}
            getBook={[Function]}
            getBookError={null}
            isFetching={false}
            match={
              Object {
                "params": Object {
                  "bookId": 97,
                },
              }
            }
            reviewBook={[Function]}
            reviews={
              Array [
                Object {
                  "bookId": 98,
                  "createdAt": "2018-07-10T08:51:25.310Z",
                  "id": 20,
                  "review": "Review",
                  "updatedAt": "2018-07-10T08:51:25.310Z",
                  "userId": 16,
                  "userReviews": Object {
                    "createdAt": "2018-07-08T19:29:20.334Z",
                    "email": "rtochi57@gmail.com",
                    "firstName": "Agbo",
                    "id": 16,
                    "imageURL": null,
                    "lastName": "Tochi",
                    "role": "User",
                    "updatedAt": "2018-07-08T19:29:20.334Z",
                  },
                },
              ]
            }
            upvoteBook={[Function]}
          />
        </div>,
      ],
      "className": "container wrapper",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <BookDetails
              book={
                Object {
                  "author": "Michelle McNamara",
                  "bookReviews": Array [
                    Object {
                      "bookId": 98,
                      "createdAt": "2018-07-10T08:51:25.310Z",
                      "id": 20,
                      "review": "Review",
                      "updatedAt": "2018-07-10T08:51:25.310Z",
                      "userId": 16,
                      "userReviews": Object {
                        "createdAt": "2018-07-08T19:29:20.334Z",
                        "email": "rtochi57@gmail.com",
                        "firstName": "Agbo",
                        "id": 16,
                        "imageURL": null,
                        "lastName": "Tochi",
                        "role": "User",
                        "updatedAt": "2018-07-08T19:29:20.334Z",
                      },
                    },
                  ],
                  "borrowCount": 6,
                  "createdAt": "2018-05-10T03:11:52.181Z",
                  "description": "A masterful true crime account of the Golden State Killer.",
                  "downvotes": 4,
                  "favCount": 3,
                  "id": 98,
                  "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                  "quantity": 3,
                  "subject": "True Crime",
                  "title": "I\\"ll Be Gone in the Dark",
                  "updatedAt": "2018-07-10T15:46:02.771Z",
                  "upvotes": 1,
                }
              }
              borrowBook={[Function]}
              favoriteBook={[Function]}
              getBook={[Function]}
              getBookError={null}
              isFetching={false}
              match={
                Object {
                  "params": Object {
                    "bookId": 97,
                  },
                }
              }
              reviewBook={[Function]}
              upvoteBook={[Function]}
            />,
            <br />,
            <Reviews
              book={
                Object {
                  "author": "Michelle McNamara",
                  "bookReviews": Array [
                    Object {
                      "bookId": 98,
                      "createdAt": "2018-07-10T08:51:25.310Z",
                      "id": 20,
                      "review": "Review",
                      "updatedAt": "2018-07-10T08:51:25.310Z",
                      "userId": 16,
                      "userReviews": Object {
                        "createdAt": "2018-07-08T19:29:20.334Z",
                        "email": "rtochi57@gmail.com",
                        "firstName": "Agbo",
                        "id": 16,
                        "imageURL": null,
                        "lastName": "Tochi",
                        "role": "User",
                        "updatedAt": "2018-07-08T19:29:20.334Z",
                      },
                    },
                  ],
                  "borrowCount": 6,
                  "createdAt": "2018-05-10T03:11:52.181Z",
                  "description": "A masterful true crime account of the Golden State Killer.",
                  "downvotes": 4,
                  "favCount": 3,
                  "id": 98,
                  "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                  "quantity": 3,
                  "subject": "True Crime",
                  "title": "I\\"ll Be Gone in the Dark",
                  "updatedAt": "2018-07-10T15:46:02.771Z",
                  "upvotes": 1,
                }
              }
              borrowBook={[Function]}
              favoriteBook={[Function]}
              getBook={[Function]}
              getBookError={null}
              isFetching={false}
              match={
                Object {
                  "params": Object {
                    "bookId": 97,
                  },
                }
              }
              reviewBook={[Function]}
              reviews={
                Array [
                  Object {
                    "bookId": 98,
                    "createdAt": "2018-07-10T08:51:25.310Z",
                    "id": 20,
                    "review": "Review",
                    "updatedAt": "2018-07-10T08:51:25.310Z",
                    "userId": 16,
                    "userReviews": Object {
                      "createdAt": "2018-07-08T19:29:20.334Z",
                      "email": "rtochi57@gmail.com",
                      "firstName": "Agbo",
                      "id": 16,
                      "imageURL": null,
                      "lastName": "Tochi",
                      "role": "User",
                      "updatedAt": "2018-07-08T19:29:20.334Z",
                    },
                  },
                ]
              }
              upvoteBook={[Function]}
            />,
          ],
          "id": "book-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "book": Object {
                "author": "Michelle McNamara",
                "bookReviews": Array [
                  Object {
                    "bookId": 98,
                    "createdAt": "2018-07-10T08:51:25.310Z",
                    "id": 20,
                    "review": "Review",
                    "updatedAt": "2018-07-10T08:51:25.310Z",
                    "userId": 16,
                    "userReviews": Object {
                      "createdAt": "2018-07-08T19:29:20.334Z",
                      "email": "rtochi57@gmail.com",
                      "firstName": "Agbo",
                      "id": 16,
                      "imageURL": null,
                      "lastName": "Tochi",
                      "role": "User",
                      "updatedAt": "2018-07-08T19:29:20.334Z",
                    },
                  },
                ],
                "borrowCount": 6,
                "createdAt": "2018-05-10T03:11:52.181Z",
                "description": "A masterful true crime account of the Golden State Killer.",
                "downvotes": 4,
                "favCount": 3,
                "id": 98,
                "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                "quantity": 3,
                "subject": "True Crime",
                "title": "I\\"ll Be Gone in the Dark",
                "updatedAt": "2018-07-10T15:46:02.771Z",
                "upvotes": 1,
              },
              "borrowBook": [Function],
              "favoriteBook": [Function],
              "getBook": [Function],
              "getBookError": null,
              "isFetching": false,
              "match": Object {
                "params": Object {
                  "bookId": 97,
                },
              },
              "reviewBook": [Function],
              "upvoteBook": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "book": Object {
                "author": "Michelle McNamara",
                "bookReviews": Array [
                  Object {
                    "bookId": 98,
                    "createdAt": "2018-07-10T08:51:25.310Z",
                    "id": 20,
                    "review": "Review",
                    "updatedAt": "2018-07-10T08:51:25.310Z",
                    "userId": 16,
                    "userReviews": Object {
                      "createdAt": "2018-07-08T19:29:20.334Z",
                      "email": "rtochi57@gmail.com",
                      "firstName": "Agbo",
                      "id": 16,
                      "imageURL": null,
                      "lastName": "Tochi",
                      "role": "User",
                      "updatedAt": "2018-07-08T19:29:20.334Z",
                    },
                  },
                ],
                "borrowCount": 6,
                "createdAt": "2018-05-10T03:11:52.181Z",
                "description": "A masterful true crime account of the Golden State Killer.",
                "downvotes": 4,
                "favCount": 3,
                "id": 98,
                "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                "quantity": 3,
                "subject": "True Crime",
                "title": "I\\"ll Be Gone in the Dark",
                "updatedAt": "2018-07-10T15:46:02.771Z",
                "upvotes": 1,
              },
              "borrowBook": [Function],
              "favoriteBook": [Function],
              "getBook": [Function],
              "getBookError": null,
              "isFetching": false,
              "match": Object {
                "params": Object {
                  "bookId": 97,
                },
              },
              "reviewBook": [Function],
              "reviews": Array [
                Object {
                  "bookId": 98,
                  "createdAt": "2018-07-10T08:51:25.310Z",
                  "id": 20,
                  "review": "Review",
                  "updatedAt": "2018-07-10T08:51:25.310Z",
                  "userId": 16,
                  "userReviews": Object {
                    "createdAt": "2018-07-08T19:29:20.334Z",
                    "email": "rtochi57@gmail.com",
                    "firstName": "Agbo",
                    "id": 16,
                    "imageURL": null,
                    "lastName": "Tochi",
                    "role": "User",
                    "updatedAt": "2018-07-08T19:29:20.334Z",
                  },
                },
              ],
              "upvoteBook": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
          <div
            id="book-details"
          >
            <BookDetails
              book={
                Object {
                  "author": "Michelle McNamara",
                  "bookReviews": Array [
                    Object {
                      "bookId": 98,
                      "createdAt": "2018-07-10T08:51:25.310Z",
                      "id": 20,
                      "review": "Review",
                      "updatedAt": "2018-07-10T08:51:25.310Z",
                      "userId": 16,
                      "userReviews": Object {
                        "createdAt": "2018-07-08T19:29:20.334Z",
                        "email": "rtochi57@gmail.com",
                        "firstName": "Agbo",
                        "id": 16,
                        "imageURL": null,
                        "lastName": "Tochi",
                        "role": "User",
                        "updatedAt": "2018-07-08T19:29:20.334Z",
                      },
                    },
                  ],
                  "borrowCount": 6,
                  "createdAt": "2018-05-10T03:11:52.181Z",
                  "description": "A masterful true crime account of the Golden State Killer.",
                  "downvotes": 4,
                  "favCount": 3,
                  "id": 98,
                  "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                  "quantity": 3,
                  "subject": "True Crime",
                  "title": "I\\"ll Be Gone in the Dark",
                  "updatedAt": "2018-07-10T15:46:02.771Z",
                  "upvotes": 1,
                }
              }
              borrowBook={[Function]}
              favoriteBook={[Function]}
              getBook={[Function]}
              getBookError={null}
              isFetching={false}
              match={
                Object {
                  "params": Object {
                    "bookId": 97,
                  },
                }
              }
              reviewBook={[Function]}
              upvoteBook={[Function]}
            />
            <br />
            <Reviews
              book={
                Object {
                  "author": "Michelle McNamara",
                  "bookReviews": Array [
                    Object {
                      "bookId": 98,
                      "createdAt": "2018-07-10T08:51:25.310Z",
                      "id": 20,
                      "review": "Review",
                      "updatedAt": "2018-07-10T08:51:25.310Z",
                      "userId": 16,
                      "userReviews": Object {
                        "createdAt": "2018-07-08T19:29:20.334Z",
                        "email": "rtochi57@gmail.com",
                        "firstName": "Agbo",
                        "id": 16,
                        "imageURL": null,
                        "lastName": "Tochi",
                        "role": "User",
                        "updatedAt": "2018-07-08T19:29:20.334Z",
                      },
                    },
                  ],
                  "borrowCount": 6,
                  "createdAt": "2018-05-10T03:11:52.181Z",
                  "description": "A masterful true crime account of the Golden State Killer.",
                  "downvotes": 4,
                  "favCount": 3,
                  "id": 98,
                  "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                  "quantity": 3,
                  "subject": "True Crime",
                  "title": "I\\"ll Be Gone in the Dark",
                  "updatedAt": "2018-07-10T15:46:02.771Z",
                  "upvotes": 1,
                }
              }
              borrowBook={[Function]}
              favoriteBook={[Function]}
              getBook={[Function]}
              getBookError={null}
              isFetching={false}
              match={
                Object {
                  "params": Object {
                    "bookId": 97,
                  },
                }
              }
              reviewBook={[Function]}
              reviews={
                Array [
                  Object {
                    "bookId": 98,
                    "createdAt": "2018-07-10T08:51:25.310Z",
                    "id": 20,
                    "review": "Review",
                    "updatedAt": "2018-07-10T08:51:25.310Z",
                    "userId": 16,
                    "userReviews": Object {
                      "createdAt": "2018-07-08T19:29:20.334Z",
                      "email": "rtochi57@gmail.com",
                      "firstName": "Agbo",
                      "id": 16,
                      "imageURL": null,
                      "lastName": "Tochi",
                      "role": "User",
                      "updatedAt": "2018-07-08T19:29:20.334Z",
                    },
                  },
                ]
              }
              upvoteBook={[Function]}
            />
          </div>,
        ],
        "className": "container wrapper",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BookDetails
                book={
                  Object {
                    "author": "Michelle McNamara",
                    "bookReviews": Array [
                      Object {
                        "bookId": 98,
                        "createdAt": "2018-07-10T08:51:25.310Z",
                        "id": 20,
                        "review": "Review",
                        "updatedAt": "2018-07-10T08:51:25.310Z",
                        "userId": 16,
                        "userReviews": Object {
                          "createdAt": "2018-07-08T19:29:20.334Z",
                          "email": "rtochi57@gmail.com",
                          "firstName": "Agbo",
                          "id": 16,
                          "imageURL": null,
                          "lastName": "Tochi",
                          "role": "User",
                          "updatedAt": "2018-07-08T19:29:20.334Z",
                        },
                      },
                    ],
                    "borrowCount": 6,
                    "createdAt": "2018-05-10T03:11:52.181Z",
                    "description": "A masterful true crime account of the Golden State Killer.",
                    "downvotes": 4,
                    "favCount": 3,
                    "id": 98,
                    "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                    "quantity": 3,
                    "subject": "True Crime",
                    "title": "I\\"ll Be Gone in the Dark",
                    "updatedAt": "2018-07-10T15:46:02.771Z",
                    "upvotes": 1,
                  }
                }
                borrowBook={[Function]}
                favoriteBook={[Function]}
                getBook={[Function]}
                getBookError={null}
                isFetching={false}
                match={
                  Object {
                    "params": Object {
                      "bookId": 97,
                    },
                  }
                }
                reviewBook={[Function]}
                upvoteBook={[Function]}
              />,
              <br />,
              <Reviews
                book={
                  Object {
                    "author": "Michelle McNamara",
                    "bookReviews": Array [
                      Object {
                        "bookId": 98,
                        "createdAt": "2018-07-10T08:51:25.310Z",
                        "id": 20,
                        "review": "Review",
                        "updatedAt": "2018-07-10T08:51:25.310Z",
                        "userId": 16,
                        "userReviews": Object {
                          "createdAt": "2018-07-08T19:29:20.334Z",
                          "email": "rtochi57@gmail.com",
                          "firstName": "Agbo",
                          "id": 16,
                          "imageURL": null,
                          "lastName": "Tochi",
                          "role": "User",
                          "updatedAt": "2018-07-08T19:29:20.334Z",
                        },
                      },
                    ],
                    "borrowCount": 6,
                    "createdAt": "2018-05-10T03:11:52.181Z",
                    "description": "A masterful true crime account of the Golden State Killer.",
                    "downvotes": 4,
                    "favCount": 3,
                    "id": 98,
                    "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                    "quantity": 3,
                    "subject": "True Crime",
                    "title": "I\\"ll Be Gone in the Dark",
                    "updatedAt": "2018-07-10T15:46:02.771Z",
                    "upvotes": 1,
                  }
                }
                borrowBook={[Function]}
                favoriteBook={[Function]}
                getBook={[Function]}
                getBookError={null}
                isFetching={false}
                match={
                  Object {
                    "params": Object {
                      "bookId": 97,
                    },
                  }
                }
                reviewBook={[Function]}
                reviews={
                  Array [
                    Object {
                      "bookId": 98,
                      "createdAt": "2018-07-10T08:51:25.310Z",
                      "id": 20,
                      "review": "Review",
                      "updatedAt": "2018-07-10T08:51:25.310Z",
                      "userId": 16,
                      "userReviews": Object {
                        "createdAt": "2018-07-08T19:29:20.334Z",
                        "email": "rtochi57@gmail.com",
                        "firstName": "Agbo",
                        "id": 16,
                        "imageURL": null,
                        "lastName": "Tochi",
                        "role": "User",
                        "updatedAt": "2018-07-08T19:29:20.334Z",
                      },
                    },
                  ]
                }
                upvoteBook={[Function]}
              />,
            ],
            "id": "book-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "book": Object {
                  "author": "Michelle McNamara",
                  "bookReviews": Array [
                    Object {
                      "bookId": 98,
                      "createdAt": "2018-07-10T08:51:25.310Z",
                      "id": 20,
                      "review": "Review",
                      "updatedAt": "2018-07-10T08:51:25.310Z",
                      "userId": 16,
                      "userReviews": Object {
                        "createdAt": "2018-07-08T19:29:20.334Z",
                        "email": "rtochi57@gmail.com",
                        "firstName": "Agbo",
                        "id": 16,
                        "imageURL": null,
                        "lastName": "Tochi",
                        "role": "User",
                        "updatedAt": "2018-07-08T19:29:20.334Z",
                      },
                    },
                  ],
                  "borrowCount": 6,
                  "createdAt": "2018-05-10T03:11:52.181Z",
                  "description": "A masterful true crime account of the Golden State Killer.",
                  "downvotes": 4,
                  "favCount": 3,
                  "id": 98,
                  "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                  "quantity": 3,
                  "subject": "True Crime",
                  "title": "I\\"ll Be Gone in the Dark",
                  "updatedAt": "2018-07-10T15:46:02.771Z",
                  "upvotes": 1,
                },
                "borrowBook": [Function],
                "favoriteBook": [Function],
                "getBook": [Function],
                "getBookError": null,
                "isFetching": false,
                "match": Object {
                  "params": Object {
                    "bookId": 97,
                  },
                },
                "reviewBook": [Function],
                "upvoteBook": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "book": Object {
                  "author": "Michelle McNamara",
                  "bookReviews": Array [
                    Object {
                      "bookId": 98,
                      "createdAt": "2018-07-10T08:51:25.310Z",
                      "id": 20,
                      "review": "Review",
                      "updatedAt": "2018-07-10T08:51:25.310Z",
                      "userId": 16,
                      "userReviews": Object {
                        "createdAt": "2018-07-08T19:29:20.334Z",
                        "email": "rtochi57@gmail.com",
                        "firstName": "Agbo",
                        "id": 16,
                        "imageURL": null,
                        "lastName": "Tochi",
                        "role": "User",
                        "updatedAt": "2018-07-08T19:29:20.334Z",
                      },
                    },
                  ],
                  "borrowCount": 6,
                  "createdAt": "2018-05-10T03:11:52.181Z",
                  "description": "A masterful true crime account of the Golden State Killer.",
                  "downvotes": 4,
                  "favCount": 3,
                  "id": 98,
                  "imageURL": "https://res/adolfo-felix-586654-unsplash_ukhitd.jpg",
                  "quantity": 3,
                  "subject": "True Crime",
                  "title": "I\\"ll Be Gone in the Dark",
                  "updatedAt": "2018-07-10T15:46:02.771Z",
                  "upvotes": 1,
                },
                "borrowBook": [Function],
                "favoriteBook": [Function],
                "getBook": [Function],
                "getBookError": null,
                "isFetching": false,
                "match": Object {
                  "params": Object {
                    "bookId": 97,
                  },
                },
                "reviewBook": [Function],
                "reviews": Array [
                  Object {
                    "bookId": 98,
                    "createdAt": "2018-07-10T08:51:25.310Z",
                    "id": 20,
                    "review": "Review",
                    "updatedAt": "2018-07-10T08:51:25.310Z",
                    "userId": 16,
                    "userReviews": Object {
                      "createdAt": "2018-07-08T19:29:20.334Z",
                      "email": "rtochi57@gmail.com",
                      "firstName": "Agbo",
                      "id": 16,
                      "imageURL": null,
                      "lastName": "Tochi",
                      "role": "User",
                      "updatedAt": "2018-07-08T19:29:20.334Z",
                    },
                  },
                ],
                "upvoteBook": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
